### PR TITLE
Singlestore as vector and storage database.

### PR DESCRIPTION
**Pull request by vishwajeetdabholkar**
Added singlestore as vector and storage database.
Added cookbook for singlestore.
I have added code to add an indexed ANN search as a commented block. can update this when we can create embeddings column with `vector` data type.

### CODE REVIEW

File: cookbook/singlestore-vector/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## singlestore Assistant
```
Comment (ashpreetbedi) [inline cookbook/singlestore-vector/README.md:1]: nit: `SingleStore`

File: phi/vectordb/singlestore/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from phi.vectordb.distance import Distance
+from phi.vectordb.singlestore.s2vectordb import singlestore
+# from phi.vectordb.singlestore.index import Ivfflat, HNSWFlat
```
Comment (ashpreetbedi) [inline phi/vectordb/singlestore/__init__.py:3]: remove commented out import and add new line

File: phi/vectordb/singlestore/s2vectordb.py [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+        #                 text(
+        #                     f"ALTER TABLE {self.schema}.{self.collection} "
+        #                     f"ADD VECTOR INDEX ({self.table.c.embedding.name}) "
+        #                     f"INDEX_OPTIONS '{index_options_json}';"
+        #                 )
+        #             )
+        #         else:
+        #             logger.warning("Table does not exist. Cannot create vector index.")
+
+        # logger.debug("==== Vector DB Optimized ====")
```
Comment (ashpreetbedi) [inline phi/vectordb/singlestore/s2vectordb.py:356]: @vishwajeetdabholkar lets remove this code

**Comment by ashpreetbedi** [comment]
@vishwajeetdabholkar requested a few changes, also please run `./scripts/format.sh` to make sure formatting/linting passes. Thanks

**Review by ashpreetbedi — CHANGES_REQUESTED** [review]

File: phi/vectordb/singlestore/s2vectordb.py (hunk 3 above)
Comment (ashpreetbedi) [reply phi/vectordb/singlestore/s2vectordb.py:356]: @vishwajeetdabholkar is this for when ANN comes out? then maybe we keep it

**Comment by vishwajeetdabholkar** [comment]
@ashpreetbedi made the requested changes and added formatting. Let me know if any other change is required.

**Review by ashpreetbedi — APPROVED** [review]
LGTM 🚢

**Pull request merged**